src/Cake.GitHub/Releases/GitHubReleaseCreator.cs

```diff
@@ -0,0 +1,216 @@
+
+            ValidateSettings(settings);
+
+            LogSettings(owner, repository, tagName, settings);
+
+            var release = await CreateNewRelease(owner: owner, repository: repository, tagName: tagName, settings: settings);
+
+            return release;
+        }
+
+
+        private void ValidateSettings(GitHubCreateReleaseSettings settings)
+        {
+            _cakeLog.Debug("Validating GitHub Release settings");
+
+            var assetNames = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+            foreach (var filePath in settings.AssetsOrEmpty)
+            {
+                var file = _fileSystem.GetFile(filePath);
+
+                if (!file.Exists)
+                    throw new FileNotFoundException($"Cannot create GitHub Release with asset '{filePath}' because the file does not exist");
+
+                var fileName = file.Path.GetFilename().ToString();
+                if (assetNames.Contains(fileName))
+                {
+                    throw new AssetConflictException($"Cannot create GitHub release with multiple assets named '{fileName}'");
+                }
+                assetNames.Add(fileName);
+            }
+        }
+
+        private void LogSettings(string owner, string repository, string tagName, GitHubCreateReleaseSettings settings)
+        {
+            const int padding = -16;
+
+            _cakeLog.Debug("Creating GitHub Release with the following settings:");
+            _cakeLog.Debug($"\t{"Owner",padding}: '{owner}'");
+            _cakeLog.Debug($"\t{"Repository",padding}: '{repository}'");
+            _cakeLog.Debug($"\t{"TagName" ,padding}: '{tagName}'");
+            _cakeLog.Debug($"\t{nameof(settings.TargetCommitish),padding}: '{settings.TargetCommitish}'");
+            _cakeLog.Debug($"\t{nameof(settings.Name),padding}: '{settings.Name}'");
+            _cakeLog.Debug($"\t{nameof(settings.Body),padding}: '{settings.Body}'");
+            _cakeLog.Debug($"\t{nameof(settings.Draft),padding}: {settings.Draft}");
+            _cakeLog.Debug($"\t{nameof(settings.Overwrite),padding}: {settings.Overwrite}");
+            _cakeLog.Debug($"\t{nameof(settings.Prerelease),padding}: {settings.Prerelease}");
+            if (settings.AssetsOrEmpty.Any())
+            {
+                _cakeLog.Debug($"\t{nameof(settings.Assets),padding}:");
+                foreach (var asset in settings.AssetsOrEmpty)
+                {
+                    _cakeLog.Debug($"\t\t- {asset}");
+                }
+            }
+        }
+
+        private async Task<GitHubRelease> CreateNewRelease(string owner, string repository, string tagName, GitHubCreateReleaseSettings settings)
+        {            
+            // 
+            // Check for existing release and delete if necessary
+            //
+            var existingRelease = await TryGetReleaseAsync(owner: owner, repository: repository, tagName: tagName);
+            if (existingRelease != null)
```

Comment:
I think "is not null" is preferred nowadays

Reply:
The `is not null` pattern requires C# 9 which won't work on the Visual Studio 2017 build machines currently used for CI.
I didn't want to change any unrelated config as part of a feature PR

Reply:
Good point.